src/tx/builder/schema.ts

```diff
@@ -317,477 +359,578 @@ export const TX_FEE_OTHER_GAS = (txType, txSize, { relativeTtl, innerTxSize }) =
+  ['calls', FIELD_TYPES.mptrees],
+  ['channels', FIELD_TYPES.mptrees],
+  ['contracts', FIELD_TYPES.mptrees],
+  ['ns', FIELD_TYPES.mptrees],
+  ['oracles', FIELD_TYPES.mptrees]
+] as const
+export interface TxProofOfInclusion extends BuildTxArgBySchema<typeof PROOF_OF_INCLUSION_TX> {}
 
 const STATE_TREES_TX = [
   ...BASE_TX,
-  TX_FIELD('contracts', FIELD_TYPES.rlpBinary),
-  TX_FIELD('calls', FIELD_TYPES.rlpBinary),
-  TX_FIELD('channels', FIELD_TYPES.rlpBinary),
-  TX_FIELD('ns', FIELD_TYPES.rlpBinary),
-  TX_FIELD('oracles', FIELD_TYPES.rlpBinary),
-  TX_FIELD('accounts', FIELD_TYPES.rlpBinary)
-]
+  ['contracts', FIELD_TYPES.rlpBinary],
+  ['calls', FIELD_TYPES.rlpBinary],
+  ['channels', FIELD_TYPES.rlpBinary],
+  ['ns', FIELD_TYPES.rlpBinary],
+  ['oracles', FIELD_TYPES.rlpBinary],
+  ['accounts', FIELD_TYPES.rlpBinary]
+] as const
+export interface TxStateTrees extends BuildTxArgBySchema<typeof STATE_TREES_TX> {}
 
 const MERKLE_PATRICIA_TREE_TX = [
   ...BASE_TX,
-  TX_FIELD('values', FIELD_TYPES.rlpBinaries)
-]
+  ['values', FIELD_TYPES.rlpBinaries]
+] as const
+export interface TxMerklePatriciaTree extends BuildTxArgBySchema<typeof MERKLE_PATRICIA_TREE_TX> {}
 
 const MERKLE_PATRICIA_TREE_VALUE_TX = [
   ...BASE_TX,
-  TX_FIELD('key', FIELD_TYPES.hex),
-  TX_FIELD('value', FIELD_TYPES.rawBinary)
-]
+  ['key', FIELD_TYPES.hex],
+  ['value', FIELD_TYPES.rawBinary]
+] as const
+export interface TxMerklePatriciaTreeValue
+  extends BuildTxArgBySchema<typeof MERKLE_PATRICIA_TREE_VALUE_TX> {}
 
 const CONTRACTS_TREE_TX = [
   ...BASE_TX,
-  TX_FIELD('contracts', FIELD_TYPES.rlpBinary)
-]
+  ['contracts', FIELD_TYPES.rlpBinary]
+] as const
+export interface TxContractsTree extends BuildTxArgBySchema<typeof CONTRACTS_TREE_TX> {}
 
 const CONTRACT_CALLS_TREE_TX = [
   ...BASE_TX,
-  TX_FIELD('calls', FIELD_TYPES.rlpBinary)
-]
+  ['calls', FIELD_TYPES.rlpBinary]
+] as const
+export interface TxContractCallsTree extends BuildTxArgBySchema<typeof CONTRACT_CALLS_TREE_TX> {}
 
 const CHANNELS_TREE_TX = [
   ...BASE_TX,
-  TX_FIELD('channels', FIELD_TYPES.rlpBinary)
-]
+  ['channels', FIELD_TYPES.rlpBinary]
+] as const
+export interface TxChannelsTree extends BuildTxArgBySchema<typeof CHANNELS_TREE_TX> {}
 
 const NAMESERVICE_TREE_TX = [
   ...BASE_TX,
-  TX_FIELD('mtree', FIELD_TYPES.rlpBinary)
-]
+  ['mtree', FIELD_TYPES.rlpBinary]
+] as const
+export interface TxNameServiceTree extends BuildTxArgBySchema<typeof NAMESERVICE_TREE_TX> {}
 
 const ORACLES_TREE_TX = [
   ...BASE_TX,
-  TX_FIELD('otree', FIELD_TYPES.rlpBinary)
-]
+  ['otree', FIELD_TYPES.rlpBinary]
+] as const
+export interface TxOraclesTree extends BuildTxArgBySchema<typeof ORACLES_TREE_TX> {}
 
 const ACCOUNTS_TREE_TX = [
   ...BASE_TX,
-  TX_FIELD('accounts', FIELD_TYPES.rlpBinary)
-]
+  ['accounts', FIELD_TYPES.rlpBinary]
+] as const
+export interface TxAccountsTree extends BuildTxArgBySchema<typeof ACCOUNTS_TREE_TX> {}
+
+export type TxParams = TxAccount | TxSpend | TxSigned
```

Comment:
I don't think you need to combine these interfaces, better to switch to specific one in particular case